clients/cobol-lsp-vscode-extension/src/test/suite/processorgroups.spec.test.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as assert from "assert";
+import * as helper from "./testHelper";
+import * as vscode from "vscode";
+import * as path from "path";
+import * as fs from "fs";
+import { getWorkspacePath } from "./testHelper";
+
+interface PgmConf {
+  program: string;
+  pgroup: string;
+}
+
+function remove_pgm_cfg(pgmConf: PgmConf) {
+  const pgm_conf_path = path.join(getWorkspacePath(), ".cobolplugin", "pgm_conf.json");
+  const fileContents = fs.readFileSync(pgm_conf_path, 'utf8');
+  const pgmConfs = JSON.parse(fileContents);
+  pgmConfs.pgms = pgmConfs.pgms.filter(
+    (obj: PgmConf) => obj.program !== pgmConf.program && obj.pgroup !== pgmConf.pgroup
+  );
+  const updatedJson = JSON.stringify(pgmConfs, null, 4);
+  fs.writeFileSync(pgm_conf_path, updatedJson, 'utf8');
+}
+
+function add_pgm_cfg(pgmConf: PgmConf) {
+  const pgm_conf_path = path.join(getWorkspacePath(), ".cobolplugin", "pgm_conf.json");
+  const fileContents = fs.readFileSync(pgm_conf_path, 'utf8');
+  const pgmConfs = JSON.parse(fileContents);
+  pgmConfs.pgms.push(pgmConf);
+  const updatedJson = JSON.stringify(pgmConfs, null, 4);
+  fs.writeFileSync(pgm_conf_path, updatedJson, 'utf8');
+}
```

Comment:
Why is this thing needed? Why not simply have it all in the configuration files to begin with?

Reply:
to simulate the existence or non existence of the processor groups

Reply:
Surely that can be accomplished without modifying the config files, like referencing a group and not providing a definition.

Reply:
If I use a group with no definition, I will need to use a group with definition I want a "no errors scenario", which means I need to modify the files

Reply:
Could you elaborate how is removing and entry from config different from having e.g. folder `cobol-idms-noconfig` without an entry in the config?

Reply:
now I see what you meant